src/SimulationMap.cu

```diff
@@ -99,6 +113,9 @@ __device__ SimulationMap::SimulationMap(Polyhedron *polyhedron) :
             create_new_nodes = false;
         }
     }
+
+    free(nodes_directions);
```

Comment:
(just a note, no need to fix anything)
I would place this line to a different commit, probably even not in this branch